kore/test/Test/Kore/Step/Rule/Common.hs

```diff
@@ -0,0 +1,114 @@
+module Test.Kore.Step.Rule.Common
+    ( Pair (..)
+    , RuleBase (..)
+    ) where
+
+import Prelude.Kore
+
+
+import qualified Data.Default as Default
+
+import qualified Kore.Internal.OrPattern as OrPattern
+import qualified Kore.Internal.Pattern as Pattern
+import Kore.Internal.Predicate
+    ( Predicate
+    , makeTruePredicate
+    , makeTruePredicate_
+    )
+import qualified Kore.Internal.Predicate as Predicate
+import Kore.Internal.TermLike
+    ( TermLike
+    , VariableName
+    )
+import qualified Kore.Internal.TermLike as TermLike
+import Kore.Rewriting.RewritingVariable
+    ( mkRuleVariable
+    )
+import Kore.Step.ClaimPattern
+    ( OnePathRule (..)
+    , claimPatternInternal
+    )
+import Kore.Step.RulePattern
+    ( RHS (RHS)
+    , RewriteRule (RewriteRule)
+    , RulePattern (RulePattern)
+    )
+import qualified Kore.Step.RulePattern as OLD
+
+newtype Pair variable = Pair (TermLike variable, Predicate variable)
+
+class RuleBase base rule where
```

Comment:
Is this class very useful, now that we can have whole `Pattern`s on the left- and right-hand sides?

Reply:
The tests in `Simplify` should work on `RewriteRule`s as well. I might remove this class once I clean up all the old code, but right now it seemed the cleanest way to run the tests with old/new claims and rewrite rules.